core/src/main/java/jenkins/slaves/DefaultJnlpSlaveReceiver.java

```diff
@@ -48,7 +48,7 @@ public boolean handle(String nodeName, JnlpSlaveHandshake handshake) throws IOEx
                 }
             } else {
                 handshake.error(nodeName + " is already connected to this master. Rejecting this connection.");
-                return false;
+                return true;
```

Comment:
Javadoc: " true if the name was claimed and the handshake was completed (either successfully or unsuccessfully)". The clause seems to be an unsuccessful handshake, so the change seems to be valid